### ISSUE TITLE
Calendar: Calendar using java.util.Date and jsf native converter crashes on conversion error

### ISSUE THREAD

**Issue by arnis87** [primefaces/primefaces#5923]
**Describe the defect**
`p:calendar ` component will crash when updating page after converter exception due to internal conversion failure of `LocalDateTime.now()`
In `CalendarRenderer`/`DatePickerRenderer `line [90 ](https://github.com/primefaces/primefaces/blob/076b68be327546432b1c033c18ddfea789f8defa/src/main/java/org/primefaces/component/calendar/CalendarRenderer.java#L90) it tries to convert `defaultDate ` using `CalendarUtils `which in turn throws `ConversionException ` on its own being unable to convert `LocalDateTime` into string since the `CalendarUtils` delegates the conversion to `DateTimeConverter` which fails to convert new java date objects.

Not sure where the bug is but at very least it should catch the potential default date conversion in Renderer as it has no control over the outcome.

**Reproducer**

**Environment:**
 - PF Version: _8.0_
 - JSF + version: _e.g. Mojarra 2.3.9 
 - Affected browsers: _e.g. Chrome, iOS, ALL_

**To Reproduce**
use `p:calendar ` component in combination with native jsf dateconverter and invoke conversion error on page. There is javascript error in console while updating the page.

**Expected behavior**
At very least - not get the error in page, resulting in page not being rendered correctly. 

**Example XHTML**
```xml
<p:calendar value="#{regularJavaUtilDateObj}">
    <f:convertDateTime />
</p:calendar>
```
**Example Java**
```java
if (calendar.isConversionFailed()) {
   defaultDate = CalendarUtils.getValueAsString(context, calendar, LocalDateTime.now());
}
```
**Workaround**
Managed to get around it by defining custom local _Renderer_ overrides for `DatePickerRenderer/CalendarRenderer ` in _faces-config.xml_ but this requires duplicating the entire `encodeScript` method. 
```java
if (calendar.isConversionFailed()) {
   defaultDate = CalendarUtils.getValueAsString(context, calendar, new Date());
}
```
The latter works in my specific situation but it might not work for everyone.

**Comment by tandraschko** [comment]
@christophs78 can you also check this one?

**Comment by melloware** [comment]
I have confirmed this issue on 9.0-SNAPSHOT as well.

**Comment by arnis87** [comment]
Would like to have this fix available in next 8.0 elite version as well
